File: Java-Core/src/main/java/presto/type/ResourceType.java
package presto.type;

import presto.runtime.Context;

public class ResourceType extends CategoryType {

	public ResourceType(String name) {
		super(name);
	}
	
	@Override
	public boolean equals(Object obj) {
		if(obj==this)
			return true;
		if(obj==null)
			return false;
		if(!(obj instanceof ResourceType))
			return false;
		ResourceType other = (ResourceType)obj;
		return this.getName().equals(other.getName());
	}
	
	@Override
	public boolean isAssignableTo(Context context, IType other) {
		return this.equals(other);
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/type/ResourceType.js
var CategoryType = require("./CategoryType").CategoryType;

function ResourceType(name) {
	CategoryType.call(this, name);
	return this;
}

ResourceType.prototype = Object.create(CategoryType.prototype);
ResourceType.prototype.constructor = ResourceType;

ResourceType.prototype.equals = function(obj) {
	if(obj==this) {
		return true;
	}
	if(!(obj instanceof ResourceType)) {
		return false;
	}
	return this.name==objname;
};

ResourceType.prototype.isAssignableTo = function(context, other) {
	return this.equals(other);
};
	
exports.ResourceType = ResourceType;
